processor/attributewhitelistprocessor/processor.go

```diff
@@ -0,0 +1,88 @@
+package attributewhitelistprocessor
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+	"regexp"
+)
+
+type attributewhitelistprocessor struct {
+	nextConsumer       consumer.TraceConsumer
+	attributeWhitelist []string
+}
+
+func newTraceProcesor(
+	params component.ProcessorCreateParams,
+	nextConsumer consumer.TraceConsumer,
+	config Config) (component.TraceProcessor, error) {
+	proc := &attributewhitelistprocessor{
+		nextConsumer:       nextConsumer,
+		attributeWhitelist: config.AttributeWhiteList,
+	}
+	return proc, nil
+}
+
+
+func (wp *attributewhitelistprocessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+	// drop tags when they don't match the white list
+	resourcespans := td.ResourceSpans()
+	for i := 0; i < resourcespans.Len(); i++ {
+		rs := resourcespans.At(i)
+		if rs.IsNil() {
+			continue
+		}
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			if ils.IsNil() {
+				continue
+			}
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				s := spans.At(k)
+				if s.IsNil() {
+					continue
+				}
+				attributes := s.Attributes()
+
+				var attributesToDelete []string
+				findAttrsToDelete := func(k string, v pdata.AttributeValue) {
+					if wp.shouldDeleteTag(k) {
+						attributesToDelete = append(attributesToDelete, k)
+					}
+				}
+				attributes.ForEach(findAttrsToDelete)
+				for _, k := range attributesToDelete {
+					attributes.Delete(k)
+				}
+			}
+		}
+	}
+	return wp.nextConsumer.ConsumeTraces(ctx, td)
+}
+
+func (wp *attributewhitelistprocessor) GetCapabilities() component.ProcessorCapabilities {
+	return component.ProcessorCapabilities{MutatesConsumedData: true}
```

Comment:
Clearly we're mutating data, but what is the side effect of setting this option?

Reply:
If my understanding is correct, it uses this flag to decide if there needs to be a fan out connector to prevent one pipeline from using the mutated data from another pipeline.

https://github.com/Zocdoc/opentelemetry-collector/blob/9d16f5971ef950b554d12db1533202e33f5b93c2/service/builder/receivers_builder.go#L280